blockly-core/core/ui/contract_editor/example_view.js

```diff
@@ -52,14 +50,19 @@ Blockly.ExampleView.prototype.initializeTestButton = function (buttonText, iconC
 };
 
 Blockly.ExampleView.prototype.testExample_ = function () {
-  this.resultText.innerHTML = this.onExampleRun_(this.block_);
+  this.contractEditor_.resetExampleViews();
```

Comment:
Could the contract editor pass in references to resetExampleViews and testResetHandler_ as parameters? Just to clarify the boundary between the view and editor in the constructor signature.